pdf/pdfcanvas.cpp

```diff
@@ -78,6 +89,12 @@ class PDFCanvas::Private
     float spacing;
 
     QRectF visibleArea;
+    bool rememberPagePosition;
+    QString dSource;
```

Comment:
Ok, I'll put full name. For the copy of document->source() in dSource, it's something I don't know in C++. Explanations :
* PDFCanvas::Private has a pointer to a document. It is set by PDFCanvas::setDocument() by just a pointer assignment (d->document = document).
* when ~PDFCanvas() is called, the actual document instance it points to has already been destroyed, so calling document->source() segfaults there.

In C with glib I would have count ref document (g_object_ref(document)) in the setDocument() method, to be sure that document is available during the full life of my PDFCanvas object. I would have called g_object_unref() then in the destroy method of PDFCanvas. I don't know how to do this in C++, *i.e.* I don't know how to add a reference to an object to be sure that we can call its methods whenever we want to.